plugin/pkg/replacer/replacer.go

```diff
@@ -14,13 +15,27 @@ import (
 )
 
 // Replacer replaces labels for values in strings.
-type Replacer struct {
-	valueFunc func(request.Request, *dnstest.Recorder, string) string
-	labels    []string
+type Replacer struct{}
+
+// New makes a new replacer. This only needs to be called once in the setup and
+// then call Replace for each incoming message. A replacer is safe for concurrent use.
+func New() Replacer {
+	return Replacer{}
 }
 
+// Replace performs a replacement of values on s and returns the string with the replaced values.
+func (r Replacer) Replace(ctx context.Context, state request.Request, rr *dnstest.Recorder, s string) string {
+	return loadFormat(s).Replace(ctx, state, rr)
+}
+
+const (
+	headerReplacer = "{>"
+	// EmptyValue is the default empty value.
+	EmptyValue = "-"
+)
+
 // labels are all supported labels that can be used in the default Replacer.
-var labels = []string{
+var labels = [...]string{
```

Comment:
why the `...` here?

Reply:
That was a nit on my part.  I like to declare what are essentially `const slices` as arrays to indicate to other programers that they are const and not to be modified.

That said, the `labels` var is no longer necessary and I will replace it with the `knownLabels` map.